### PR TITLE
Add swagger configuration

### PR DESCRIPTION
**Pull request by Bhavesh-SP18**
- Add basic swagger configuration.
- Add.gitignore.
- Update the readme.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -21,3 +21,97 @@
...
+.idea/**/sqlDataSources.xml
+.idea/**/dynamic.xml
+.idea/**/uiDesigner.xml
+.idea/**/dbnavigator.xml
+
+# Gradle
+.idea/**/gradle.xml
+.idea/**/libraries
+
+# Gradle and Maven with auto-import
```
Comment (kunalcodess) [inline .gitignore:70]: remove unused code

File: mvnw [hunk 2]
```diff
@@ -0,0 +1,316 @@
+#!/bin/sh
+# ----------------------------------------------------------------------------
```
Comment (kunalcodess) [inline mvnw:2]: this file is not needed in git history

File: mvnw.cmd [hunk 3]
```diff
@@ -0,0 +1,188 @@
+@REM ----------------------------------------------------------------------------
```
Comment (kunalcodess) [inline mvnw.cmd:1]: Is this file needed to be pushed?

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+		<version>2.5.2</version>
+		<relativePath/> <!-- lookup parent from repository -->
+	</parent>
+	<groupId>com.passbook.sparkeighteen</groupId>
+	<artifactId>passbook</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
+	<name>Passbook</name>
+	<description>Basic entry passbook</description>
+	<properties>
+		<java.version>11</java.version>
```
Comment (kunalcodess) [inline pom.xml:17]: try to use latest, i.e 17

File: src/main/java/com/passbook/sparkeighteen/config/SwaggerConfig.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+                .apis(RequestHandlerSelectors.withClassAnnotation(RestController.class))
+                .paths(PathSelectors.any())
+                .build();
+    }
+
+    private ApiInfo getApiInfo() {
+
+        return new ApiInfoBuilder()
+                .title("Swagger API Doc")
+                .description("More description about the API")
```
Comment (kunalcodess) [inline src/main/java/com/passbook/sparkeighteen/config/SwaggerConfig.java:33]: Can be something else

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -0,0 +1,7 @@
+spring.datasource.url=jdbc:postgresql://localhost:5432/passbook
+spring.datasource.username=postgres
+spring.datasource.password=qwerty1201
+spring.jpa.hibernate.ddl-auto=create-drop
```
Comment (kunalcodess) [inline src/main/resources/application.properties:4]: do not use ddl to create tables

**Review by kunalcodess — CHANGES_REQUESTED** [review]

File: src/main/java/com/passbook/sparkeighteen/config/SwaggerConfig.java (hunk 5 above)
Comment (Aashishverma06) [reply src/main/java/com/passbook/sparkeighteen/config/SwaggerConfig.java:33]: No sir swagger configure krne ke bad ka msg hai bas.
Screenshot 2022-06-13 at 11 57 03 AM

Comment (spark-coders) [inline src/main/java/com/passbook/sparkeighteen/config/SwaggerConfig.java:33]: @Aashishverma06 
He is asking you to update the description line which make sense to end user who will be consuming your api.

File: src/main/java/com/passbook/sparkeighteen/peristence/entity/UserEntity.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+import javax.persistence.*;
+import java.util.UUID;
+
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity
+@Table(name = "user",schema = "public")
```
Comment (spark-coders) [inline src/main/java/com/passbook/sparkeighteen/peristence/entity/UserEntity.java:14]: Why was this added please let me know.

File: src/main/java/com/passbook/sparkeighteen/peristence/repository/UserRepository.java [hunk 8]
```diff
@@ -0,0 +1,10 @@
+package com.passbook.sparkeighteen.peristence.repository;
+
+import com.passbook.sparkeighteen.peristence.entity.UserEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.Optional;
+
+public interface UserRepository  extends JpaRepository<UserEntity, Integer> {
+    Optional<UserEntity> findByEmail(String email);
```
Comment (spark-coders) [inline src/main/java/com/passbook/sparkeighteen/peristence/repository/UserRepository.java:9]: Why do we use optional let me know.

File: src/main/java/com/passbook/sparkeighteen/service/UserService.java [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+
+import java.util.Optional;
+
+@Service
+public class UserService {
+
+    private UserRepository userRepository;
+
+    public UserService(final UserRepository userRepository) {
```
Comment (spark-coders) [inline src/main/java/com/passbook/sparkeighteen/service/UserService.java:18]: Why did we use constructor injection.
What are the alternative injection strategy.

File: src/main/java/com/passbook/sparkeighteen/service/UserService.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    public SignUpResponse signUp(@NonNull final SignUpRequest signUpRequest) throws Exception {
+        final Optional<UserEntity> byEmail =
+                userRepository.findByEmail(signUpRequest.getEmail());
+        if (byEmail.isPresent()) {
+            return SignUpResponse.builder()
+                    .message("Email already exists.")
+                    .build();
+        }
```
Comment (spark-coders) [inline src/main/java/com/passbook/sparkeighteen/service/UserService.java:29]: What does the above code block does.

File: src/main/java/com/passbook/sparkeighteen/service/UserService.java [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+import com.passbook.sparkeighteen.peristence.POJO.SignUpResponse;
+import com.passbook.sparkeighteen.peristence.entity.UserEntity;
+import com.passbook.sparkeighteen.peristence.repository.UserRepository;
+import lombok.NonNull;
+import org.springframework.stereotype.Service;
+
+
+import java.util.Optional;
+
+@Service
```
Comment (spark-coders) [inline src/main/java/com/passbook/sparkeighteen/service/UserService.java:13]: Why did we write @Service and not @Componnet

File: src/main/java/com/passbook/sparkeighteen/peristence/POJO/SignUpRequest.java [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+
+import javax.validation.constraints.NotEmpty;
+
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class SignUpRequest {
+
+    @NotEmpty(message = "email is mandatory")
```
Comment (spark-coders) [inline src/main/java/com/passbook/sparkeighteen/peristence/POJO/SignUpRequest.java:16]: Can you explain the significance of @Empty

File: src/main/java/com/passbook/sparkeighteen/controller/UserController.java [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+
+    private UserService userService;
+
+    public UserController(final UserService userService){
+        this.userService = userService;
+    }
+
+    @ApiOperation("sign-up the user to passbook")
+    @PostMapping("/sign-up")
+    public ResponseEntity<SignUpResponse> signUp(@Valid @RequestBody final SignUpRequest signUpRequest) throws Exception {
```
Comment (spark-coders) [inline src/main/java/com/passbook/sparkeighteen/controller/UserController.java:27]: Why is the return type ResponseEntity?

File: src/main/java/com/passbook/sparkeighteen/config/SwaggerConfig.java [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+import springfox.documentation.builders.RequestHandlerSelectors;
+import springfox.documentation.service.ApiInfo;
+import springfox.documentation.spi.DocumentationType;
+import springfox.documentation.spring.web.plugins.Docket;
+import springfox.documentation.swagger2.annotations.EnableSwagger2;
+
+@Configuration
+@EnableSwagger2
+public class SwaggerConfig {
+    @Bean
```
Comment (spark-coders) [inline src/main/java/com/passbook/sparkeighteen/config/SwaggerConfig.java:19]: Why did we use @Bean here any significance?

File: .gitignore (hunk 1 above)
Comment (Ketan-SP18) [reply .gitignore]: removed

File: mvnw (hunk 2 above)
Comment (Ketan-SP18) [reply mvnw:2]: Removed

File: mvnw.cmd (hunk 3 above)
Comment (Ketan-SP18) [reply mvnw.cmd:1]: file added to gitignore

File: pom.xml (hunk 4 above)
Comment (Ketan-SP18) [reply pom.xml:17]: Yes, using 17

File: src/main/resources/application.properties (hunk 6 above)
Comment (Ketan-SP18) [reply src/main/resources/application.properties]: yes commented that

File: src/main/java/com/passbook/sparkeighteen/controller/UserController.java (hunk 13 above)
Comment (Ketan-SP18) [reply src/main/java/com/passbook/sparkeighteen/controller/UserController.java:27]: ResponseEntity allows to control http status code that is returned

File: src/main/java/com/passbook/sparkeighteen/peristence/entity/UserEntity.java (hunk 7 above)
Comment (Bhavesh-SP18) [reply src/main/java/com/passbook/sparkeighteen/peristence/entity/UserEntity.java:14]: @Entity annotation specifies that the class is an entity and is mapped to a database table.

File: src/main/java/com/passbook/sparkeighteen/peristence/POJO/SignUpRequest.java (hunk 12 above)
Comment (Ketan-SP18) [reply src/main/java/com/passbook/sparkeighteen/peristence/POJO/SignUpRequest.java:16]: @NotEmpty - Valid as long as its not null and its size is greater than zero.

File: src/main/java/com/passbook/sparkeighteen/peristence/repository/UserRepository.java (hunk 8 above)
Comment (Bhavesh-SP18) [reply src/main/java/com/passbook/sparkeighteen/peristence/repository/UserRepository.java:9]: **Optional**  is java 8 feature and it is a container object which may or may not contain a non-null value.If a value is present, isPresent() will return true and get() will return the value

File: mvnw.cmd (hunk 3 above)
Comment (Hrishikesh-SP18) [reply mvnw.cmd:1]: mvnw.cmd added to gitignore

File: pom.xml (hunk 4 above)
Comment (Hrishikesh-SP18) [reply pom.xml:17]: update java version 11 to 17

File: src/main/java/com/passbook/sparkeighteen/controller/UserController.java (hunk 13 above)
Comment (kunalcodess) [reply src/main/java/com/passbook/sparkeighteen/controller/UserController.java:27]: @Ketan-SP18 status code is not the only reason, please look into this more and come back with concrete explanation

File: src/main/java/com/passbook/sparkeighteen/service/UserService.java (hunk 9 above)
Comment (Aashishverma06) [reply src/main/java/com/passbook/sparkeighteen/service/UserService.java:18]: It is a type of Spring Dependency Injection, where object's constructor is used to inject dependencies. This type of injection is safer as the objects won't get created if the dependencies aren't available or dependencies cannot be resolved.

Alternatives injection 
1. Constructor-Based Injection
2. Setter Injection
3. Property-Based Injection

File: src/main/java/com/passbook/sparkeighteen/service/UserService.java (hunk 11 above)
Comment (Hrishikesh-SP18) [reply src/main/java/com/passbook/sparkeighteen/service/UserService.java:13]: - In an application, the business logic resides within the service layer so we use the @Service Annotation to indicate that a class belongs to that layer. 
- It is also a specialization of @Component Annotation like the @Repository Annotation. 
- One most important thing about the @Service Annotation is it can be applied only to classes. 
- It is used to mark the class as a service provider. 
- So overall @Service annotation is used with classes that provide some business functionalities. 
- Spring context will auto detect these classes when annotation-based configuration and class-path scanning is used.

File: src/main/java/com/passbook/sparkeighteen/config/SwaggerConfig.java (hunk 14 above)
Comment (Hrishikesh-SP18) [reply src/main/java/com/passbook/sparkeighteen/config/SwaggerConfig.java:19]: - Spring @Bean Annotation is applied on a method to specify that it returns a bean to be managed by Spring context.
- Bean annotation is usually declared in Configuration classes methods.

File: src/main/java/com/passbook/sparkeighteen/config/SwaggerConfig.java (hunk 5 above)
Comment (Bhavesh-SP18) [reply src/main/java/com/passbook/sparkeighteen/config/SwaggerConfig.java:33]: PASSBOOK APPLICATION DOCUMENTATION

File: src/main/java/com/passbook/sparkeighteen/service/UserService.java (hunk 10 above)
Comment (spark-coders) [reply src/main/java/com/passbook/sparkeighteen/service/UserService.java:29]: @Bhavesh-SP18 @Aashishverma06 @Ketan-SP18 @Hrishikesh-SP18 
Reply soon will merger this PR soon.

File: src/main/java/com/passbook/sparkeighteen/service/UserService.java [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+public class UserService {
+
+    private UserRepository userRepository;
+
+    public UserService(final UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    public SignUpResponse signUp(@NonNull final SignUpRequest signUpRequest) throws Exception {
+        final Optional<UserEntity> byEmail =
```
Comment (kunalcodess) [inline src/main/java/com/passbook/sparkeighteen/service/UserService.java:23]: change variable name - As suggested do not use auto-generated variable name

**Review by kunalcodess — CHANGES_REQUESTED** [review]

**Review by kunalcodess — APPROVED** [review]
